tinyagentos/routes/desktop_browser/proxy.py

```
@@ -1,84 +1,226 @@
+    response: httpx.Response | None = None
+    # Sentinel for SSRF blocks detected inside the inner coroutine.
+    _ssrf_blocked_url: list[str] = []
+    _too_many_redirects: list[bool] = []
+
+    async def _fetch_with_redirects() -> httpx.Response | None:
+        nonlocal current_url
+        _resp: httpx.Response | None = None
+        async with httpx.AsyncClient(
+            follow_redirects=False, timeout=_HOP_TIMEOUT,
+        ) as http:
+            for hop in range(_MAX_REDIRECTS + 1):
```

Ruff (0.15.12)

[warning] Loop control variable `hop` not used within loop body

Rename unused `hop` to `_hop`

(B007)